### PR TITLE
MGMT-9495: Remove unused isoeditor functions and isoutil package

### PR DESCRIPTION
**Pull request by carbonin**
After removing the iso functions from s3wrapper these are now unused and
can be removed.

## List all the issues related to this PR

Part 3 of https://issues.redhat.com/browse/MGMT-9495
Built on https://github.com/openshift/assisted-service/pull/3688

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @nmagnezi 
/cc @danielerez 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/isoeditor/rhcos.go [hunk 1]
```diff
@@ -237,22 +64,17 @@ func RamdiskImageArchive(netFiles []staticnetworkconfig.StaticNetworkConfigData,
...
 	}
 
-	return getCompressedArchive(buffer)
-}
-
-func addCustomRAMDisk(clusterISOPath string, netFiles []staticnetworkconfig.StaticNetworkConfigData, clusterProxyInfo *ClusterProxyInfo, ramdiskOffsetInfo *OffsetInfo) error {
-	compressedArchive, err := RamdiskImageArchive(netFiles, clusterProxyInfo)
-	if err != nil {
-		return err
+	// Run gzip compression
```
Comment (nmagnezi) [inline internal/isoeditor/rhcos.go:67]: Just making sure, this is coming to replace the functionality you removed with `getCompressedArchive()`, right?

**Review by nmagnezi — COMMENTED** [review]
overall looks good. Just one question in the comments.

File: internal/isoeditor/rhcos.go (hunk 1 above)
Comment (carbonin) [reply internal/isoeditor/rhcos.go:67]: Yup, the function would have only been called from this one place after the changes so I just inlined it here.

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**